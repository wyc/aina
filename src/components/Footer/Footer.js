import React from 'react';

import { Link } from 'gatsby';

const Footer = () => {
  const date = new Date().getFullYear();
  const copyright = `Â© ${date} Victoria H. Chang`;

  return (
    <footer className='Footer'>
      <Link to='/'>Home</Link>
      <div className='Footer__copyright'>{copyright}</div>
    </footer>
  );
};

export default Footer;
