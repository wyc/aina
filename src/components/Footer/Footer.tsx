import React, { ReactElement, ReactNode } from 'react';

export interface IFooterProps {
  children?: ReactNode;
}

const Footer = ({ children }: IFooterProps): ReactElement => {
  return (
    <footer className='Nav Row Footer'>
      {children}

      <div className='Row Footer__copy'>
        Made by{' '}
        <a
          className='Nav__link'
          href={'https://www.github.com/victoriahchang'}
          rel='noopener noreferrer'
          target='_blank'
        >
          Vi
        </a>
        with
        <div className='Emoji__heart'>ğŸ’–</div>
      </div>
    </footer>
  );
};

export default Footer;
