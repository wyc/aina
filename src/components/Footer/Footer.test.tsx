import React from 'react';
import * as renderer from 'react-test-renderer';

import Footer from './Footer';

describe('Footer', () => {
  it('renders correctly', () => {
    const props = {};
    const component = renderer.create(<Footer {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
