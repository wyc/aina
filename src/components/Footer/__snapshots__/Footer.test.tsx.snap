// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  className="Nav Row Footer"
>
  <div
    className="Row Footer__copy"
  >
    Made by
     
    <a
      className="Nav__link"
      href="https://www.github.com/victoriahchang"
      rel="noopener noreferrer"
      target="_blank"
    >
      Vi
    </a>
    with
    <div
      className="Emoji__heart"
    >
      ðŸ’–
    </div>
  </div>
</footer>
`;
